h/models/group.py

```diff
@@ -16,6 +18,8 @@
 GROUP_NAME_MIN_LENGTH = 3
 GROUP_NAME_MAX_LENGTH = 25
 GROUP_DESCRIPTION_MAX_LENGTH = 250
+AUTHORITY_PROVIDED_ID_PATTERN = "^[a-zA-Z0-9._\-+!~*()']+$"
```

Comment:
I wonder if we actually need this restriction or if we could just require identifiers to be percent-encoded with `encodeURIComponent`, `urllib.quote` etc. if they may contain anything that is not allowed. I suppose we can always relax this constraint later.